src/py42/_internal/clients/alertrules.py

```diff
@@ -0,0 +1,161 @@
+        uri = u"{0}{1}".format(self._api_prefix, u"remove-all-users")
+        return self._session.post(uri, data=json.dumps(data))
+
+    def get(self, rule_type, rule_id):
+        """Fetch alert rules by rule id.
+
+        Args:
+            rule_id (str): Id of a rule to be updated.
+            rule_type (str): Either of 'exfiltration', 'cloudshare', 'typemismatch' where
+              'exfiltration' implies rules related to movement of files,
+              'cloudshare' implies rules related to sharing/providing public access to files
+              'typemismatch' implies rules related to content and file extension mismatch.
+
+        Returns
+            :class:`py42.response.Py42Response`
+        """
+        tenant_id = self._user_context.get_current_tenant_id()
+        data = {"tenantId": tenant_id, "ruleIds": [rule_id]}
+        uri = u"{0}{1}".format(self._api_prefix, _ALERT_RULE_ENDPOINTS[rule_type])
+        return self._session.post(uri, data=json.dumps(data))
+
+
+class AlertRulesManagerClient(BaseClient):
+    """A client to manage Alert Rules."""
+
+    _version = u"v1"
+    _resource = u"rules/"
+    _api_prefix = u"/svc/api/{0}/{1}".format(_version, _resource)
+
+    def __init__(self, session, user_context):
+        super(AlertRulesManagerClient, self).__init__(session)
+        self._user_context = user_context
+
+    def _get_alert_rules(
+        self, tenant_id, sort_key=None, sort_direction=None, page_num=None, page_size=None
+    ):
+        data = {
+            "tenantId": tenant_id,
+            "groups": [],
+            "groupClause": "AND",
+            "pgNum": page_num,
+            "pgSize": page_size,
+            "srtKey": sort_key,
+            "srtDirection": sort_direction,
+        }
+        uri = u"{0}{1}".format(self._api_prefix, u"query-rule-metadata")
+        return self._session.post(uri, data=json.dumps(data))
+
+    def get_all(self, sort_key="CreatedAt", sort_direction="DESC"):
+        """Fetch all available rules.
+
+        Args:
+            sort_key (str): Sort results based by field, default 'CreatedAt'.
+            sort_direction (str): ``ASC`` or ``DESC``, default "DESC"
+
+        Returns:
+            generator: An object that iterates over :class:`py42.response.Py42Response` objects
+            that each contain a page of events.
+        """
+        tenant_id = self._user_context.get_current_tenant_id()
+        return get_all_pages(
+            self._get_alert_rules,
+            u"ruleMetadata",
+            tenant_id=tenant_id,
+            sort_key=sort_key,
+            sort_direction=sort_direction,
+        )
+
+    def get_by_name(self, rule_name):
+        """Fetch a rule by its name.
+
+        Args:
+            rule_name (str): Name to search for, case insensitive search.
+
+        Returns
+            :class:`py42.response.Py42Response`
+        """
+        rules = self.get_all()
+
+        for rule in rules:
```

Comment:
See https://py42docs.code42.com/en/stable/userguides/basics.html#paging if you get stuck here

Reply:
Thank you, In the latest changes, I am sending a json response instead of Py42Response, is that fine?

Reply:
I think the `get_by_name()` function / logc only needs to be in the Alerts Rules Module and not this internal client. I think it makes sense for a module to do this sort of logic using clients, which are more expected to return Py42Responses.